recengine/recengine.py

```diff
@@ -104,20 +104,22 @@ def _create_program_from_csv_reader(program_reader):
     program = defaultdict(list)
     previous_set = None
     current_day_index = 0
+    first_datetime = None
     for row in program_reader:
```

Comment:
I think some narrative comments inside the for-loop would help with understanding what's going on in this function. Also maybe add a docstring to the function. This is nitpicking though.